// Copyright 2018 The Chromium Authors. All rights reserved.
// Use of this source code is govered by a BSD-style
// license that can be found in the LICENSE file or at
// https://developers.google.com/open-source/licenses/bsd

syntax = "proto3";


package monorail;

import "api/api_proto/common.proto";
import "api/api_proto/project_objects.proto";


service Projects {
  rpc ListProjects (ListProjectsRequest) returns (ListProjectsResponse) {}
  rpc GetConfig (GetConfigRequest) returns (Config) {}
}


message ListProjectsRequest {
  int32 page_size = 1;
  string page_token = 2;
}


message ListProjectsResponse {
  repeated Project projects = 1;
  string next_page_token = 2;
}


// Next available tag: 3
message GetConfigRequest {
  RequestTrace trace = 1;
  string project_name = 2;
}
